api/nodes/views.py

```diff
@@ -92,17 +98,60 @@ class NodeDetail(generics.RetrieveUpdateAPIView, NodeMixin):
         ContributorOrPublic,
         ReadOnlyIfRegistration,
     )
-    serializer_class = NodeSerializer
 
-    # overrides RetrieveUpdateAPIView
+    def get_serializer_class(self):
+        if self.request.method == 'DELETE':
+            serializer_class = NodeDeleteSerializer
+            return serializer_class
+        serializer_class = NodeSerializer
+        return serializer_class
+
+    # overrides RetrieveUpdateDestroyAPIView
     def get_object(self):
         return self.get_node()
 
-    # overrides RetrieveUpdateAPIView
+    # overrides RetrieveUpdateDestroyAPIView
     def get_serializer_context(self):
         # Serializer needs the request in order to make an update to privacy
         return {'request': self.request}
 
+    def delete(self, request, node_id):
+        user = self.request.user
+        node = self.get_node()
+        token = node_token_creator(node._id, user._id)
+        url = absolute_reverse('nodes:node-delete-confirm', kwargs={'node_id': node._id, 'token': token})
+        delete_warning = BEFORE_DELETE_NODE.format(_(node.title))
+        return Response({'data': {'node_id': node._id, 'warning_message': delete_warning, 'links': {'confirm_delete': url}}}, status=status.HTTP_202_ACCEPTED)
+
+
+class NodeDeleteConfirm(generics.DestroyAPIView, NodeMixin):
+    """Projects and component details.
+
+    On the front end, nodes are considered 'projects' or 'components'. The difference between a project and a component
+    is that a project is the top-level node, and components are children of the project. There is also a category field
+    that includes the option of project. The categorization essentially determines which icon is displayed by the
+    Node in the front-end UI and helps with search organization. Top-level Nodes may have a category other than
+    project, and children nodes may have a category of project.
+    """
+    permission_classes = (
+        ContributorOrPublic,
```

Comment:
If it's not admin-only, ContributorOrPublic should work, because it distinguishes the type of access method and responds appropriately.

```py
        if request.method in permissions.SAFE_METHODS:
            return obj.is_public or obj.can_view(auth)
        else:
            return obj.can_edit(auth)
```

Safe methods are read only. The rest require write permission.